src/coreComponents/meshUtilities/CPMesh/utilities/GeometryUtilities.hpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file GeometryUtilities.hpp
+ */
+
+#ifndef GEOSX_MESHUTILITIES_CPMESH_UTILITIES_GEOMETRYUTILITIES_HPP_
+#define GEOSX_MESHUTILITIES_CPMESH_UTILITIES_GEOMETRYUTILITIES_HPP_
+
+namespace geosx
+{
+
+namespace CPMesh
+{
+
+namespace GeometryUtilities
+{
+
+/**
+ * @struct Vertex
+ * @brief a helper struct used to filter out duplicate nodes
+ */
+struct Vertex
```

Comment:
I'd be surprised we do not have this already. Good for starting nevertheless.

Reply:
There is definitely the same thing in Pamela, but I don't think there is something like that inside GEOSX.